### PR TITLE
fix: Validate search vector dimensions before PQ decoding

### PR DESCRIPTION
**Pull request by parkerduckworth**
### What's being changed:

With a PQ-enabled collection, performing a vector search with incorrect dimensionality results in a panic in the quantizer, during distance calculations with the compressed vectors. 

This PR adds a validation check before beginning comparison of the search vector with the stored compressed vectors, returning an error rather than panicking. 

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/repos/db/vector/hnsw/delete_test.go [hunk 1]
```diff
@@ -951,18 +951,18 @@ func TestDelete_ResetLockDoesNotLockForever(t *testing.T) {
...
 			err := vectorIndex.Delete(vectorIndex.entryPointID)
-			ellapsed = time.Since(starting)
+			elapsed = time.Since(starting)
 			require.Nil(t, err)
 		}()
 		wg.Wait()
-		fmt.Println(ellapsed.Milliseconds())
-		assert.LessOrEqual(t, ellapsed.Milliseconds(), int64(10))
+		fmt.Printf("ms elapsed: %d\n", elapsed.Milliseconds())
+		assert.LessOrEqual(t, elapsed.Milliseconds(), int64(15))
```
Comment (parkerduckworth) [inline adapters/repos/db/vector/hnsw/delete_test.go:965]: @abdelr we have seen this test flake quite a bit with elapsed ms of `11`, `13`, and so forth. Haven't seen an instance where it is higher than 13, so I changed this max elapsed value to `15`

Comment (antas-marcin) [reply adapters/repos/db/vector/hnsw/delete_test.go:965]: I think that Abdel made a [PR](https://github.com/weaviate/weaviate/pull/5985/files) with a fix for this test targeted for `stable/v1.24`

Comment (parkerduckworth) [reply adapters/repos/db/vector/hnsw/delete_test.go:965]: Ah very nice. Would it make sense to keep this, and allow it to be overridden with the merge-forward? This way the improved logging can stay?

Comment (antas-marcin) [reply adapters/repos/db/vector/hnsw/delete_test.go:965]: let's do that!

**Review by antas-marcin — DISMISSED** [review]

**Review by dirkkul — COMMENTED** [review]
Can you add a (unit-)test for this?

**Comment by parkerduckworth** [comment]
@dirkkul added unit + acceptance tests

**Review by antas-marcin — APPROVED** [review]

**Review by antas-marcin — DISMISSED** [review]

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**